otcextensions/sdk/css/v1/snapshot.py

```diff
@@ -0,0 +1,153 @@
+    #: Indices that need to be backed up.
+    indices = resource.Body('indices')
+    #: Total number of shards of the indices to be backed up.
+    total_shards = resource.Body('totalShards', type=int)
+    #: Number of shards that fail to be backed up.
+    failed_shards = resource.Body('failedShards', type=int)
+    #: Version of the snapshot.
+    version = resource.Body('version')
+    #: Snapshot restoration status.
+    restore_status = resource.Body('restoreStatus')
+    #: Timestamp when the snapshot starts to be executed.
+    start_time = resource.Body('startTime')
+    #: Timestamp when the snapshot execution ends.
+    end_time = resource.Body('endTime')
+    #: Bucket for storing snapshot data.
+    bucket_name = resource.Body('bucketName')
+
+    #: ID of the cluster, to which the snapshot is to be restored.
+    target_cluster = resource.Body('targetCluster')
+    #: Name of the index to be restored.
+    indices = resource.Body('indices')
+    #: Rule for defining the indices to be restored.
+    rename_pattern = resource.Body('renamePattern')
+    #: Rule for renaming an index.
+    rename_replacement = resource.Body('renameReplacement')
+
+    def restore(self, session, **body):
+        """Preform actions given the message body.
+        """
+        url = utils.urljoin(self.base_path, self.id, 'restore')
+        return session.post(url, json=body)
+
+
+class SnapshotPolicy(resource.Resource):
+
+    base_path = '/clusters/%(cluster_id)s/index_snapshot/policy'
+
+    allow_create = True
+    allow_fetch = True
+
+    #: ID of the cluster where automatic snapshot creation is enabled.
+    cluster_id = resource.URI('cluster_id')
+
+    #: Retention days for a snapshot.
+    backup_keep_days = resource.Body('keepday', type=int)
+    #: Time when a snapshot is created every day.
+    backup_period = resource.Body('period')
+    #: Snapshot name prefix.
+    backup_prefix = resource.Body('prefix')
+    #: OBS bucket for storing snapshots.
+    bucket_name = resource.Body('bucket')
+    #: Storage path of the snapshot in the OBS bucket.
+    backup_path = resource.Body('basePath')
+    #: Agency used to access OBS buckets.
+    agency = resource.Body('agency')
+    #: Whether to enable the automatic snapshot creation policy.
+    enable = resource.Body('enable', type=otc_format.BoolStr_1)
+    #: Name of the index to be backed up.
+    indices = resource.Body('indices')
+    #: Snapshot encryption ID.
+    cmk_id = resource.Body('snapshotCmkId')
+    #: Whether to delete all automatically created snapshots when the
+    #:  automatic snapshot creation policy is disabled.
+    delete_auto = resource.Body('deleteAuto')
+
+
+class SnapshotConfiguration(resource.Resource):
+
+    base_path = '/clusters/%(cluster_id)s/index_snapshot/%(setting)s'
+
+    allow_create = True
+
+    #: ID of the cluster where automatic snapshot creation is enabled.
+    clusterId = resource.URI('cluster_id')
```

Comment:
cluster_id is used for resource.Body

Reply:
I checked the API description and couldn't find the second clusterId parameter. So from my POV it does not make sense to use or can you show me where the second one is necessary.

Reply:
the snapshot list API returns clusterId - https://docs.otc.t-systems.com/api/css/css_03_0034.html
and for snapshotUrl it also needs cluster_id, but one is resource.Body and another is resource.URI

Reply:
okay understand. Nevertheless I would not mix it up as it is right now, that cluster_id is clusterId and vise versa. So that at the end cluster_id is cluster_id and clusterId is clusterId ^^.

Reply:
We have discussed this issue and Vineet had the opinion to let the user faced variable be 'cluster_id' for path and output and the hidden one to 'clusterId' which makes sense. So I close and confirm the current state.

Reply:
he? but why would you need ever to know the clusterId from the resource if you already have cluster_id from url? This way will only create mess. Simply ignore whatever is in the body

Reply:
ok, will be ignored in the next push. we are talking right now.